### ISSUE TITLE
searchByAttributes return all user when checking the equality of a custom attribute against an empty string

### ISSUE THREAD

**Issue by maxime-bouchee** [keycloak/keycloak#31811]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/client-java

### Describe the bug

Hello,
We have an issue when we use the searchByAttributes method. When checking the equality of a custom attribute against an empty string, we found a User even if there is no user validating the query.
Here is an example of the issue :
```
 realm.users()
            .searchByAttributes(
                0, 1, true, false, "custom_attribut:"""); 
```
The result is the first userRepresentation that has the attribut "custom_attribut" and not the one that has custom_attribut equals to empty string.

Is it a normal behavior ?

### Version

24.0.5

### Regression

- [ ] The issue is a regression

### Expected behavior

When checking the equality of a custom attribute against an empty string, we found a User validating the query

### Actual behavior

When checking the equality of a custom attribute against an empty string, we found a User even if there are no user validating the query

### How to Reproduce?

 realm.users()
            .searchByAttributes(
                0, 1, true, false, "custom_attribut:"""); with no user having this attribute equals to empty string

**Comment by maxime-bouchee** [comment]
We reproduce the same issue in v25 of keycloak

**Comment by vramik** [comment]
@maxime-bouchee I agree that returning all users is not exactly ideal. Keycloak uses `SearchQueryUtils` class to parse map of search attributes from string: https://github.com/keycloak/keycloak/blob/242796b89679933ae8df63953960308eef5afb07/services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java#L277-L279

and the empty value is evaluated as empty map which leads to getting everything.

We discussed the issue in our internal call and turned out it's unclear what would be expected outcome of calling the `realm.users().searchByAttributes(0, 1, true, false, "custom_attribut:""");` ... would it be expected all users who has the `custom_attribut` set with any value or all users with the `custom_attribut` unset?

**Comment by maxime-bouchee** [comment]
Thank you for your answer :+1: 
If the request is valid for you, it should return an empty array, if it is not, I think you should throw an error or just return an error code 404.

**Comment by vramik** [comment]
Thank you @maxime-bouchee for the reply. I was wondering why did you send the request in the first place.

I am opening the issue as enhancement. To me it makes sense to return empty list.

**Comment by maxime-bouchee** [comment]
Thank you @vramik. We sent this request by mistake. We forget to clean some data we use to request keycloak trhought an admin swagger and there was some empty string in our request that's why :)
